### PR TITLE
Improve phylo_trees#index performance

### PR DESCRIPTION
**Pull request by cdebourcy**
# Description
- This PR fixes performance regressions due to inefficient `TaxonLineage` and `MetadataField` queries. 
- A follow-up PR should fix performance problems due to the fact that all metadata for all phylo_trees are being loaded, just to display the tree list. We should have separate `index` and `show` actions, where `index` just gets tree names and such, whereas `show` gets all the info for a single tree, only if the user actually visits that tree.
- A review of the performance of all metadata-related actions across the entire app will be necessary in the future, since a number of inefficient queries have crept in.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix a feature that knowingly causes existing functionality to not work as expected)

## Before
Excerpts from `log/development.log` of 1 single `phylo_trees#index` action:
screen shot 2019-02-27 at 2 32 52 pm
...
screen shot 2019-02-27 at 2 33 01 pm
...
screen shot 2019-02-27 at 2 33 24 pm
...

## After
only O(1) query per table (`log/development.log`)
screen shot 2019-02-27 at 2 28 26 pm

### CODE REVIEW

File: app/models/metadatum.rb [hunk 1]
```diff
@@ -235,6 +235,21 @@ def validated_value
     ""
   end
 
+  def self.validated_value_multiget(metadata)
+    metadata_fields = MetadataField.where(id: metadata.pluck(:metadata_field_id)).index_by(&:id)
```
Comment (jshoe) [inline app/models/metadatum.rb]: you could also do .map(&:metadata_field) inside

Comment (cdebourcy) [reply app/models/metadatum.rb]: You mean like `metadata_fields = metadata.map(&:metadata_field).index_by(&:id)`? That would incur a query for each metadata record.
image
image

Comment (jshoe) [reply app/models/metadatum.rb]: Ah ok, never mind then

Comment (cdebourcy) [reply app/models/metadatum.rb]: cool

**Pull request merged**